Update Dockerfile to install Node.js to tier3 runtime image

pxpeterxu opened the pull request:
### Motivation

Related to https://github.com/DMOJ/judge-server/pull/1143:

After enabling support in judge-server for a Node.js-based executor, we need to install the pre-requisites in the Docker image used for the judge.

### Changes

- Install the latest version of Node.js on Debian using Nodesource repositories

### Testing

Build image with updated Dockerfile, and ensured that Node.js was properly installed:

```sh
# Ran this on Ubuntu, but any platform with Docker installed should work
cd runtimes-docker/tier3
sudo docker build --tag=dmoj/runtimes-tier3 .
sudo docker run --entrypoint node -it dmoj/runtimes-tier3 --version
# <-- output: v20.10.0
```

Xyene commented:
This looks good to me (but we can drop the comments), thanks! Could you please also rebase to resolve the merge conflicts that have been picked up?

After that we can merge this, and CI should start passing on the judge-server PR.

pxpeterxu committed:
Install Node.js on tier3 runtime image

pxpeterxu commented:
> This looks good to me (but we can drop the comments), thanks! Could you please also rebase to resolve the merge conflicts that have been picked up?
> 
> After that we can merge this, and CI should start passing on the judge-server PR.

Just rebased and removed comments around adding Nodesource repository.

Xyene reviewed (approved):
LGTM, thanks!
